ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -89,7 +91,10 @@
   private StoreFindToken recoveryEndToken;
   private HardDeletePersistInfo hardDeleteRecoveryRange = new HardDeletePersistInfo();
   private Throttler throttler;
-  boolean isCaughtUp = false;
+  private boolean isCaughtUp = false;
+  private final AtomicBoolean paused = new AtomicBoolean(false);
+  private final ReentrantLock reentrantLock = new ReentrantLock();
```

Comment:
this is used for many purposes. Like for timed wait while waiting after being caught up and indefinite wait while in pause state. Not sure if we can have a single apt name to reflect both. May be "sharedLock" or something. Can you think of something.